idb/postgres/postgres.go

```diff
@@ -1052,24 +1052,26 @@ ON CONFLICT (addr, assetid) DO UPDATE SET amount = account_asset.amount + EXCLUD
 
 					// Asset Config
 					if au.Config != nil {
-						var outparams []byte
+						var params types.AssetParamsWithExtra
 						if au.Config.IsNew {
-							outparams = encoding.EncodeJSON(au.Config.Params)
+							params = au.Config.Params
 						} else {
 							row := getacfg.QueryRow(au.AssetID)
 							var paramjson []byte
 							err = row.Scan(&paramjson)
 							if err != nil {
 								return fmt.Errorf("get acfg %d, %v", au.AssetID, err)
 							}
-							var old sdk_types.AssetParams
+							var old types.AssetParamsWithExtra
 							err = encoding.DecodeJSON(paramjson, &old)
 							if err != nil {
 								return fmt.Errorf("bad acgf json %d, %v", au.AssetID, err)
 							}
-							np := types.MergeAssetConfig(old, au.Config.Params)
-							outparams = encoding.EncodeJSON(np)
+							new := au.Config.Params
+							params = types.MergeAssetConfig(old, new)
 						}
+						params = encoding.ConvertAssetParams(params)
```

Comment:
I think that the conversion here should be from `types.AssetParams` -> `types.AssetParamsWithExtra` and not from `AssetParamsWithExtra` -> `AssetParamsWithExtra`

Reply:
I shifted around the scope a bit to clean this up.

`AssetParamsWithExtra` is now part of the encoding layer.

The idb/api layers share the model object for sharing the string/byte fields (this is already shared in a couple places for other model objects).